.github/workflows/build_wheels.yml

```diff
@@ -26,7 +26,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-20.04, macos-10.15] #, windows-2019]
-        pyver: ["3.6", "3.7", "3.8", "3.9", "3.10"]
+        pyver: ["3.6", "3.8", "3.9", "3.10"]
```

Comment:
@tommy-waltmann @klywang This change is awkward because it means we'll stop providing wheels for Python 3.7 but continue providing Python 3.6 wheels... Hmm. A gap like that is unusual/unexpected. The Python 3.7 cibuildwheel job fails at the _testing_ stage (after the wheel is built) when it installs the pinned dependencies from [requirements/requirements-test.txt](https://github.com/glotzerlab/freud/blob/master/requirements/requirements-test.txt). CI output here: https://github.com/glotzerlab/freud/runs/4680663466

I suggest one of the following courses of action:
- Add a new requirements file that is not pinned to "newest" but instead uses `>=min_required_version` for all dependencies. It would be an exact clone of [.circleci/ci-oldest-reqs.txt](https://github.com/glotzerlab/freud/blob/master/.circleci/ci-oldest-reqs.txt) but with `>=` instead of `==`. We could call it `requirements/requirements-test-compatible.txt` or something like that. Dependabot won't update `>=` pinnings so it wouldn't be changed like the other requirements files. This file's pinnings would be used for Python 3.7 (or more generally, any Python versions that are dropped by NEP 29 but we still wish to test using compatible-but-not-oldest dependency versions).
- End freud's support for Python 3.6 and 3.7, in accordance with NEP 29. We would update our "oldest" pinnings to be the lowest versions supported by Python 3.8 (we shouldn't eliminate the "oldest" CI because it ensures compatibility with older NumPy and CMake versions, among other dependencies).
- Drop cibuildwheel _tests_ for Python 3.7 but keep the CI job - cibuildwheel would only _build_ the wheel, and we would rely on the other (newest, oldest) Python versions to ensure that the wheels are adequately tested.

Reply:
Of those three options, I think the first is the best way to proceed. I think many people are still using python 3.6/3.7, so dropping support for those versions isn't ideal. It also tests all the versions it makes wheels for, which is important for ensuring correctness.

Another option we have here is to just make python 3.7 freud's oldest supported version instead of python 3.6 and test python 3.7 wheels against `ci-oldest-reqs.txt`.

Reply:
I agree with your assessment. But I'm happy with any of the options - feel free to move forward with whichever option you like!

Reply:
I chose option number 1, we can keep supporting python3.6 for a while longer.